Extend the velbus documentation (#42862)
home-assistant/home-assistant.github.io: source/_integrations/velbus.markdown

{% endnote %}

### Binary sensor

Every input module will be represented as a binary sensor entity in Home Assistant. The binary sensor will be on when the input is active and off when the input is inactive.

### Select

Every module that supports programs will have a select entity in Home Assistant. You can select the program that you want to activate.

### Climate

Every module that supports thermostat functionality will be represented as a climate entity in Home Assistant. You can set the target temperature, the mode (heat/cool) and the preset mode (away/comfort/eco/home).

### Cover

Every blind module will be represented as a cover entity in Home Assistant. You can open and close the blind and if the module supports it, you can set the position of the blind.

### Button

Every input module will have a button entity in Home Assistant to trigger a button press event. This can be used to trigger Velbus actions from Home Assistant.

### Sensor

Some modules expose sensors that can be used in Home Assistant.
Known and implemented sensors are:

> - energy consumption sensor

Reviewer: could you add the untit, where applicable?

- temperature sensor
- light level sensor
- PSU voltage sensor
- PSU current sensor
- PSU load sensor

The Analog input module (VMB4AN) will have 4 sensor entities to read the analog values.

> The VMBMETEO module will have multiple sensor entities to read temperature, humidity, wind speed, wind direction and rain level.

Reviewer: ```suggestion
The VMBMETEO module will have multiple sensor entities to read temperature, humidity, wind speed, wind direction, and rain level.
```

  customize:
    sensor.eau_counter:
      device_class: water
```

The `device_class` attribute can have 2 values:

- gas: if the counter represents a gas meter
- water: if the counter represents a water meter

## Data updates

The integration is event driven, meaning that updates are pushed from the Velbus modules to Home Assistant as soon as a change occurs. This ensures that the state of the entities in Home Assistant is always up to date.

For sensor entities, Velbus modules send periodic updates. The interval of these updates is configurable via VelbusLink software.

## Known limitations

- The integration will never reprogram Velbus modules. All changes that are not implemented in the [Velbus module protocol](https://github.com/velbus/moduleprotocol) need to be done via VelbusLink software.
- The integration does not support upgrading Velbus modules firmware. This needs to be done via VelbusLink software.
- For relay modules, the integration does not support the forced on/off state. This needs to be done via VelbusLink software.

## Troubleshooting

If you encounter issues with the Velbus integration, you can enable debug logging to get more information about the problem.

> ### Enabling debug logging

Reviewer: ```suggestion
### Enabling debug logging

```

The Velbus integration debug log can be enabled via the graphical interface or by adding the following lines to your {% term "`configuration.yaml`" %} file:

```yaml
logger:
  default: warning
  logs:
    homeassistant.components.velbus: debug
```

After enabling debug logging, restart Home Assistant and check the logs for any error messages related to the Velbus integration.

### Can't connect to Velbus interface

> 

Reviewer: For this section, could you give a bit more detail and use the "symptom, description (optional), resolution" pattern? Example here: https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L167

For sensor entities, Velbus modules send periodic updates. The interval of these updates is configurable via VelbusLink software.

## Known limitations

- The integration will never reprogram Velbus modules. All changes that are not implemented in the [Velbus module protocol](https://github.com/velbus/moduleprotocol) need to be done via VelbusLink software.
- The integration does not support upgrading Velbus modules firmware. This needs to be done via VelbusLink software.
- For relay modules, the integration does not support the forced on/off state. This needs to be done via VelbusLink software.

## Troubleshooting

If you encounter issues with the Velbus integration, you can enable debug logging to get more information about the problem.

### Common issues

- If you have changed the configuration of your Velbus modules using VelbusLink software, you may need to clear the Velbus cache using the `velbus.clear_cache` action to ensure that Home Assistant has the latest configuration.
- If you use the VLP file configuration for the integration, ensure that the VLP file is up to date and matches the configuration of your Velbus modules.

### Enabling debug logging

> The Velbus integration debug log can be enabled via the graphical interface or by adding the following lines to your {% term "`configuration.yaml`" %} file:

Reviewer: You could also refer to the documentation on [enabling debug logs and diagnostics](https://www.home-assistant.io/docs/configuration/troubleshooting/#debug-logs-and-diagnostics).

```yaml
logger:
  default: warning
  logs:
    homeassistant.components.velbus: debug
```

After enabling debug logging, restart Home Assistant and check the logs for any error messages related to the Velbus integration.

### Can't connect to Velbus interface

#### Symptom: “Unable to setup Velbus”

This error indicates that Home Assistant is unable to connect to the Velbus interface.

#### Description

This is likely caused by one of the following issues:
- For USB connections:
  - The Velbus USB interface is not properly connected to the Home Assistant device.
  - Another application (like VelbusLink) is using the Velbus USB interface at the same time.
- For TCP/IP connections:
  - The Velbus TCP/IP interface is not powered on or not connected to the network.
  - The IP address or port number is incorrect.

>   - A firewall or network security settings are blocking the connection to the Velbus TCP/IP interface

Reviewer: ```suggestion
#### Resolution

Try the following steps:
- For USB connections:
  - Make sure the Velbus USB interface is properly connected to the Home Assistant device.
  - Make sure no other application (like VelbusLink) is using the Velbus USB interface at the same time.
- For TCP/IP connections:
  - Make sure the Velbus TCP/IP interface is powered on and connected to the network.
  - Make sure the IP address and the port number are correct.
  - Make sure the firewall or network security settings are not blocking the connection to the Velbus TCP/IP interface
```
in this case, I think the description can be skipped, as there is no new information.

#### Resolution

- For USB connections, follow the steps in the "Connection via USB" section.
- For TCP/IP connections, follow the steps in the "Connection via TCP/IP" section.

> 

Reviewer: ```suggestion
```

### Module not detected

#### Symptom: "Certain modules are not detected by the integration"

This issue indicates that some Velbus modules are not detected by the Velbus integration.

#### Description

This is likely caused by one of the following issues:
- The Velbus modules are not properly connected to the Velbus bus.
- The Velbus modules are not powered on.
- The Velbus bus scan has not been performed after adding new modules.
- The Velbus modules are not detected by the VelbusLink software.
- There are error messages in the Velbus integration debug logs related to module detection.

#### Resolution

> - Chek that Velbuslink software detects all modules correctly.

Reviewer: ```suggestion
- Check if Velbuslink software detects all modules correctly.
```

> - If this is the case file a bug report including debug logs, diagnostics and your VLP file.

Reviewer: ```suggestion
- If the Velbuslink software detects all modules, but they do not show in Home Assistant, file a bug report including debug logs, diagnostics, and your VLP file.
```
I'm not sure I understand this sorrectly. Is this what is meant here?
Another question: What is the VLP file?

Author: a VLP file is the config file from velbuslink
